DakshChan/IBS
# [IBS-12] Aggregated Grades Page

- Implemented a new page for instructors/admins where they can view all students' grades per assignment


### Steps for testing
0. Use `make build` and `make run` to build and run the docker containers. On Postman, make sure `base_url` variable is set to `http://localhost/api`. Make sure you have at least one user in DB of role `admin` and `instructor`.
1. From `http://localhost` on web browser, login as `admin` user.
    Ensure at least one course has been added to the database. Otherwise, add one right now.
   1. Create and and new students to the database through `Add role` card on `http://localhost/admin/course/1/task`:
       ```
       Username: student1
       Role: student
       Email: student1@mail.com
       Update user info: false
       ```
   2. Go to `user_info` table on database, and change the password hash values to the hash of `pass`.
   3. From Postman, login as `Instructor`. Create a new task (you can create more than 1) with the following values:
       ```
       task: task1
       due_date: 2023-06-06 12:30:45
       hidden: false
       min_member: 1
       max_member: 3
       max_token: 2
       change_group: true
       hide_interview: false
       ```
   4. Through `POST Add criteria` endpoint on Postman, send a request to backend with the following values:
       ```
       task: task1
       criteria: criteria1
       total: 50
       description: anything here
       ```
   5. To submit a mark for a student, login as an instructor on Postman, and send using the following values to `POST Submit mark` endpoint:
       ```
       task: task1
       criteria: criteria1
       username: student1
       mark: 40
       overwrite: false
       ```
   6. Logged in as an instructor on frontend, head to the following URL: `http://localhost/instructor/all-grades/:courseId`. e.g. to `http://localhost/instructor/all-grades/1`.
   7. You should see a table of all students enrolled in the course, their respective assignments as columns and the students' marks as a table cell in the respective assignment columns.
2. Examples:
    <img width="1710" alt="Screenshot 2023-05-29 at 11 43 37 AM" 
    src="https://github.com/DakshChan/IBS/assets/47423409/3226842c-f845-4856-b8fd-c87db9ef48b6">
